openapi/crmaas/crmass.json

```
@@ -0,0 +1,243 @@
+                      "type": "string"
+                    },
+                    "title": {
+                      "type": "string"
+                    },
+                    "description": {
+                      "type": "string"
+                    }
+                  },
+                  "x-examples": {
+                    "Example 1": {
+                      "id": "system__company_last_activity_date",
+                      "externalId": "system__company_last_activity_date",
+                      "type": "FIELD_TYPE_DATETIME",
+                      "title": "Last activity date",
+                      "description": "Date and time of last activity"
+                    }
+                  }
+                },
+                "examples": {
+                  "Get User Alice Smith": {
+                    "value": {
+                      "id": "FieldID-xxxx-xxxx",
+                      "externalId": "externalId",
+                      "type": "FIELD_TYPE_xxxx",
+                      "title": "Last activity date",
+                      "description": "Date and time of last activity"
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "User Not Found"
+          }
+        },
+        "operationId": "get-field-schema",
+        "description": "Retrieve field schema based on namespace."
+      }
+    },
+    "/{namespace}/{resourceTypeCode}/meta/fields/{fieldId}/options": {
+      "parameters": [
+        {
+          "schema": {
+            "type": "string"
+          },
+          "name": "namespace",
+          "in": "path",
+          "required": true,
+          "description": "The namespace, PID/AGID"
+        },
+        {
+          "schema": {
+            "type": "string"
+          },
+          "name": "resourceTypeCode",
+          "in": "path",
+          "required": true,
+          "description": "The resourceTypeCode, here Contact"
+        },
+        {
+          "schema": {
+            "type": "string"
+          },
+          "name": "fieldId",
+          "in": "path",
+          "required": true,
+          "description": "Unique identifier for field"
+        }
+      ],
+      "get": {
+        "summary": "Get field options",
+        "tags": [],
+        "responses": {
+          "200": {
+            "description": "Schema found",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object",
+                  "x-examples": {
+                    "Example 1": {
+                      "options": [
+                        "CRM"
+                      ]
+                    }
+                  },
+                  "properties": {
+                    "options": {
+                      "type": "array",
+                      "items": {
+                        "type": "string"
+                      }
```

While the internal API currently only returns a single string, I would like the external API to return an object containing the value and display name. That way we can turn IDs into something user friendly.

```suggestion
                      "items": {
                        "type": "object",
                        "properties": {
                          "value": {
                            "type": "string",
                            "description": "The value to be saved in the database."
                          },
                          "name": {
                            "type": "string",
                            "description": "The text to be displayed to users. \nNote this may be translated based on the [Accept-Language](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language) request header."
                          }
                        }
                      }
```